lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskStatus.cs

```diff
@@ -246,12 +252,20 @@ public TaskStatusProto ToProto()
                 }
                 else if (_lastException.IsPresent())
                 {
+                    byte[] error;
+                    try
+                    {
+                        error = SerializeException(_lastException.Value);
+                    }
+                    catch (SerializationException)
+                    {
+                        error = SerializeException(new NonSerializableTaskException(_lastException.Value.ToString()));
```

Comment:
Sorry, I spoke Java in a C# code review :) I meant the inner exception contained in `_lastException`, if there is any. If we can serialize that, it would be helpful to do so. Or, more generally speaking, we could check the whole stack of inner exceptions and replace the ones that can't be serialized with `NonSerializableTaskException`. Then, we serialize that new stack of exceptions. This way, preserve as much information as possible.